### ISSUE TITLE
Startup Consistency Issues

### ISSUE THREAD

**Issue by MikeHart85** [ISISComputingGroup/lewis#70]
There are a few consistency issues with the current startup process that I think should be addressed.

Some of these could be addressed separately, others are perhaps better done as a single PR.
### Adapter vs Protocol

We have an "adapters" folder which is to contain implementations for a variety of protocols. For example it contains an "epics.py" to implement the EPICS adapter.

But the startup arguments refer to "epics" as a "protocol", not an adapter. Even though the input to that "protocol" parameter is used to load "epics.py" in the "adapters" folder.

Additionally, it also introduces an "adapter" parameter that refers to the name of the class defined inside the adapter module. For example, "EpicsAdapter", which is defined in "epics.py". Do we really need to expose this? Will we ever define multiple adapters in a single adapter file? What benefit would this have over one adapter per "adapters/*.py" file?
### `import_adapter` vs other `import_*`s

`import_adapter` returns a Class reference, that then needs to be instantiated.

The others imports return an instance directly.

The `import_*`s and `get_available_submodules` are also free-standing functions that are scattered across various `__init__.py` files and `utils.py`. I think it would be better to group them together, if possible, perhaps in a central factory class as part of core.
### ArgumentParser vs Ad Hoc `--parameters`

Plankton parameters are parsed using ArgumentParser, which provides a standard `-x "short version"` / `--thisisthe "long version"` parameter interface to the user, along with friendly usage and error reporting.

Parameters to the adapter, however, have to be passed via a "--parameters" parameter, and in a "key=value" format... and, naturally, are parsed in a completely different way. Which leads to any incorrect input throwing ugly stack traces instead of friendly error reports.

I think a better solution would be to have the ArgumentParser in `simulation.py` parse what it can, and pass everything left over along to the adapter exactly as it is. The adapter should then use an ArgumentParser of its own to parse the rest. To explicitly split plankton parameters from adapter parameters, a `--` could be used.

For example:
`$ python simulation.py -d chopper -p epics --prefix SIM:`

Or, using a `--` separator and having the PV Prefix as a positional argument instead:
`$ python simulation.py -d chopper -p epics -- SIM:`
### Should be able to execute `simulation.py`

Always having to pass it to python is cumbersome and unnecessary.
### Device vs Setup

A device cannot be started without a setup or vice versa.

Specifying a device without a setup, we can reasonable default to the "default" setup.

But specifying only a setup without a device doesn't make sense at all. Currently we default to `chopper` as a convenience, but once we have dozens of device this will no longer be reasonable. 

Additionally, launching a simulation with neither a device nor a setup certainly doesn't make any sense.

Arguably, device and setup could be merged into a single parameter and, being required for every execution, it could be the primary positional parameter. Device and setup name could be separated by a `.`, `:`, `/` or similar. When omitted, setup could still default to "default". E.g.:

```
$ ./simulation.py chopper -p epics -- SIM1: &
$ ./simulation.py chopper.broken -p epics -- SIM2: &
```

Or, since all the default setup does is instantiate the device, we could get rid of the need to have a setup / default.py for that at all. Plankton could simply instantiate the device via default constructor when no setup is given.

**Comment by MichaelWedel** [comment]
I think these are valid concerns, some details still have to be thought through though (defaults etc.). This could be an umbrella issue for several small changes that would be easier to review and agree on.

**Comment by MikeHart85** [comment]
Yeah, I think so too... none of these are really burning issues. Just wanted to document them so we can keep an eye out for a good time to deal with them / have a place to discuss how we want to handle them.

**Comment by MikeHart85** [comment]
`simulation.py` was made executable in #72.

Another issue discovered in #72 is that not passing any `--parameters` causes a crash on startup. This should be fixed in #71.

**Comment by MichaelWedel** [comment]
I think this can be closed now, we have addressed all of these issues in 1.0. If any of you disagrees, please re-open.
